deps/rabbit/src/rabbit_msg_store.erl

```diff
@@ -1537,6 +1537,39 @@ scan_data(<<Size:64, MsgIdAndMsg:Size/binary, 255, Rest/bits>> = Data,
                     scan_data(Rest2, Fd, Fun, Offset + 1, FileSize, MsgIdsFound, Acc)
             end
     end;
+%% Large message alone in its own file
+scan_data(<<Size:64, MsgIdInt:128, _Rest/bits>> = Data, Fd, Fun, Offset, FileSize, _MsgIdsFound, _Acc)
+  when Size >= ?LARGE_MESSAGE_THRESHOLD + 16,
```

Comment:
(maybe the check for `LARGE_MESSAGE_THRESHOLD` is not even necessary, if `Offset == 0` and `FileSize == Size + 9` this means that this message is alone in this file and can be optimised. And the fact that the previous close did not match means this cannot be a very small message)

Reply:
Let's drop the unnecessary check.